codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -496,19 +520,49 @@ private void incrementPendingBytes(int numBytes) {
              */
             int write(int allowedBytes) {
```

Comment:
@nmittler 
this implementation should also handle the case correctly where write modifies the stream state (CLOSE / HALF_CLOSE), as then cancel() be called and the canceled flag gets set.